### PR TITLE
Introduce PerformanceProfile resource.

### PR DESCRIPTION
**Pull request by yossisegev**
Introduce PerformanceProfile resource.

### CODE REVIEW

File: ocp_resources/performance_profile.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        cpu (obj): Set of CPU related parameters. For example:
+            {
+                "isolated": "4-39,44-79",
+                "reserved": "0-3,40-43",
+            }
+        globally_disable_irq_load_balancing (bool): Toggles whether IRQ load balancing will be disabled for the Isolated CPU
+            set.  Defaults to "false".
+        hugepages (obj): Defines a set of huge pages related parameters. It is possible to set huge pages with multiple
+            size values at the same time.
+        machine_config_label (obj): Defines the label to add to the MachineConfigs the operator creates.
```
Comment (rnetser) [inline ocp_resources/performance_profile.py:20]: what is `obj`? what type should be passed here?
Do not repeat the API doc, this is why the code has aref to it.

File: ocp_resources/performance_profile.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            It has to be used in the MachineConfigSelector of the MachineConfigPool which targets this performance profile.
+            Defaults to "machineconfiguration.openshift.io/role=<same role as in NodeSelector label key>"
```
Comment (rnetser) [inline ocp_resources/performance_profile.py:22]: this is done by the api, not the code here; please remove.
the docstring should include only what is handled in the code

File: ocp_resources/performance_profile.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        cpu=None,
+        globally_disable_irq_load_balancing=False,
+        hugepages=None,
+        machine_config_label=None,
+        machine_config_pool_selector=None,
+        net=None,
+        node_selector=None,
+        numa=None,
+        workload_hints=None,
+        **kwargs,
```
Comment (rnetser) [inline ocp_resources/performance_profile.py:47]: `realTimeKernel` is missing

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/performance_profile.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            It has to be used in the MachineConfigSelector of the MachineConfigPool which targets this performance profile.
+            Defaults to "machineconfiguration.openshift.io/role=<same role as in NodeSelector label key>"
+        machine_config_pool_selector (obj): Defines the MachineConfigPool label to use in the MachineConfigPoolSelector of
+            resources like KubeletConfigs created by the operator. Defaults to
+            "machineconfiguration.openshift.io/role=<same role as in NodeSelector label key>"
+        net (obj): Defines a set of network related features.
+        numa (obj): Defines options related to topology aware affinities.
+        real_time_kernel (obj):  Defines a set of real time kernel related parameters. RT kernel won’t be installed when
+            not set.
+        workload_hints (obj): Defines hints for different types of workloads.
```
[the remark this answers is not in the document]
Comment (yossisegev) [reply ocp_resources/performance_profile.py:30]: Done

File: ocp_resources/performance_profile.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    https://docs.openshift.com/container-platform/4.13/rest_api/node_apis/performanceprofile-performance-openshift-io-v2.html#performanceprofile-performance-openshift-io-v2
+
+    Args (in .spec):
+        additional_kernel_args (array of strings): Additional kernel arguments.
+        cpu (obj): Set of CPU related parameters. For example:
+            {
+                "isolated": "4-39,44-79",
+                "reserved": "0-3,40-43",
+            }
+        globally_disable_irq_load_balancing (bool): Toggles whether IRQ load balancing will be disabled for the Isolated CPU
```
[the remark this answers is not in the document]
Comment (yossisegev) [reply ocp_resources/performance_profile.py:16]: Done

File: ocp_resources/performance_profile.py (hunk 1 above)
Comment (yossisegev) [reply ocp_resources/performance_profile.py:20]: Done

File: ocp_resources/performance_profile.py (hunk 2 above)
Comment (yossisegev) [reply ocp_resources/performance_profile.py:22]: Done

File: ocp_resources/performance_profile.py (hunk 3 above)
Comment (yossisegev) [reply ocp_resources/performance_profile.py:47]: Thanks for noticing that.
Done.

File: ocp_resources/performance_profile.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        numa=None,
+        real_time_kernel=None,
+        workload_hints=None,
+        **kwargs,
+    ):
+        """
+        PerformanceProfile object. API reference:
+        https://docs.openshift.com/container-platform/4.13/rest_api/node_apis/performanceprofile-performance-openshift-io-v2.html#performanceprofile-performance-openshift-io-v2
+
+        Args (in .spec):
```
Comment (rnetser) [inline ocp_resources/performance_profile.py:26]:
```suggestion
        Args:
```

File: ocp_resources/performance_profile.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            additional_kernel_args (array of strings): Additional kernel arguments.
```
Comment (rnetser) [inline ocp_resources/performance_profile.py:27]: what does `array of strings` mean? a list?

File: ocp_resources/performance_profile.py [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+            cpu (dict): Set of CPU related parameters. For example:
```
Comment (rnetser) [inline ocp_resources/performance_profile.py:28]: should indicate what are the expected key-value (please add in all places)

File: ocp_resources/performance_profile.py [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+                {
+                    "isolated": "4-39,44-79",
+                    "reserved": "0-3,40-43",
+                }
+            globally_disable_irq_load_balancing (bool, default: False): Toggles whether IRQ load balancing will be
+                disabled for the Isolated CPU set.
+            hugepages (dict): Defines a set of huge pages related parameters. It is possible to set huge pages with
```
Comment (rnetser) [inline ocp_resources/performance_profile.py:35]: `t is possible to set huge pages with ...` is not relevant here; the code does not do that; this is related to the api. please add what key-value are expected

File: ocp_resources/performance_profile.py [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+                multiple size values at the same time.
+            machine_config_label (dict): Defines the label to add to the MachineConfigs the operator creates.
+                It has to be used in the MachineConfigSelector of the MachineConfigPool which targets this performance
+                profile.
```
Comment (rnetser) [inline ocp_resources/performance_profile.py:39]: please remove; not related to this code

**Review by rnetser — CHANGES_REQUESTED** [review]

**Pull request closed without merging**